File: Assets/Scripts/AnimationRotateByTime.cs

// Hannes Helmholz
//
// nothing to do here anymore
using Holoville.HOTween;
using UnityEngine;

public class AnimationRotateByTime : AnimationRotateAbstract
{

    // =============================================================================
    // MEMBERS ---------------------------------------------------------------------
    [SerializeField]
    protected float RotationTime = 10f;
    // =============================================================================
 
 
 
    // =============================================================================
    // METHODS ---------------------------------------------------------------------
 
    protected override void DoFixedUpdate ()
    {
        float x = ( RotationAxis == Axis.X ) ? 1 : 0;
        float y = ( RotationAxis == Axis.Y ) ? 1 : 0;
        float z = ( RotationAxis == Axis.Z ) ? 1 : 0;
        float step = Time.fixedDeltaTime*360/-RotationTime;
     
        Trans.Rotate ( x*step, y*step, z*step );
    }
    // =============================================================================
}

File: ISS Emergency/Assets/Scripts/AnimationRotateByTime.js
#pragma strict

// Hannes Helmholz
//
// nothing to do here anymore


private class AnimationRotateByTime extends AnimationRotateAbstract {

	// =============================================================================
	// MEMBERS ---------------------------------------------------------------------
	@SerializeField	protected var RotationTime:float = 10;
	// =============================================================================
	
	
	
	// =============================================================================
	// METHODS ---------------------------------------------------------------------
	
	protected function DoFixedUpdate() {
		var x:float = (RotationAxis == Axis.X) ? 1 : 0;
		var y:float = (RotationAxis == Axis.Y) ? 1 : 0;
		var z:float = (RotationAxis == Axis.Z) ? 1 : 0;
		var step:float = Time.fixedDeltaTime * 360.0 / -RotationTime;
		
		Trans.Rotate(x * step, y * step, z * step);
	}
	// =============================================================================
}